src/store.c

```diff
@@ -683,30 +669,30 @@ pthreads_storage* pthreads_store_create(zval *unstore){
 			if (Z_TYPE_P(unstore) == IS_ARRAY)
 				storage->exists = zend_hash_num_elements(Z_ARRVAL_P(unstore));
 		} break;
+		default:
+			ZEND_ASSERT(0);
 
 	}
 	return storage;
 }
 /* }}} */
 
 /* {{{ */
-int pthreads_store_convert(pthreads_storage *storage, zval *pzval){
+void pthreads_store_save_zval(zval *zstorage, zval *write) {
+	if (!Z_REFCOUNTED_P(write)) {
```

Comment:
I'm not totally sure if this is safe. Not being refcounted doesn't necessarily mean that the value can't be destroyed.
I noticed that `zend_empty_array` is communicated without copying because of this check, which is OK in this case, but there may be other cases where this is a problem.

Reply:
In fact, this would almost definitely have problems with request-local interned strings (potential UAF).